src/PowerWAFInterface.cpp

```diff
@@ -148,8 +150,21 @@ extern "C"
         }
         try
         {
+            // TODO: make the collector optional
+            optional_ref<ddwaf::metrics_collector> mc { std::nullopt };
+            if (collector != nullptr)
+            {
+                mc = *reinterpret_cast<ddwaf::metrics_collector*>(collector);
```

Comment:
this probably breaks strict aliasing rules, and could be avoided by having `ifdef __cplusplus` in the headers to make `ddwaf_metrics_collector = ddwaf::metrics_collector*`. Another option is to use the may_alias attribute.

Reply:
The former might be okay, but the latter doesn't seem to have an alternative on MSVC.

Reply:
I've changed it to use an ifdef, it's not pretty but it avoids all the reinterpret_cast shenanigans.